lang/python/python-fnv-hash-fast/Makefile

```diff
@@ -0,0 +1,38 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-fnv-hash-fast
+PKG_VERSION:=0.5.0
+PKG_RELEASE:=1
+
+PYPI_NAME:=fnv-hash-fast
+PYPI_SOURCE_NAME:=fnv_hash_fast
+PKG_HASH:=a84d658952776a186418f4158fc8e55ff3c576ac32cc9ef7f8077efdf2d0b89f
+
+PKG_MAINTAINER:=Timothy Ace <openwrt@timothyace.com>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_BUILD_DEPENDS:=python-cython/host python-poetry-core/host
+
+include ../pypi.mk
+include $(INCLUDE_DIR)/package.mk
+include ../python3-package.mk
+
+define Package/python3-fnv-hash-fast
+  SECTION:=lang
+  CATEGORY:=Languages
+  SUBMENU:=Python
+  TITLE:=A fast version of fnv1a
+  URL:=https://github.com/bdraco/fnv-hash-fast
+  DEPENDS:=+libstdcpp +python3-light
```

Comment:
It appears that fnvhash is an [explicit dependency](https://github.com/bdraco/fnv-hash-fast/blob/v0.5.0/pyproject.toml#L30).

Reply:
@jefferyto Looking for guidance here. The [README.md](https://github.com/bdraco/fnv-hash-fast/blob/v0.5.0/README.md) states:

> This library will use a CPP implementation of fnv1a (32) if cython is available, and will fallback to pure python from the fnvhash package if it is not.

The way I read it, since we're building the CPP implementation of fnv1a (32), the module should never be falling back to the pure python polyfill from `fnvhash`. Therefore `fnvhash` is not a true dependency.

I have no issue with creating a new package for `python3-fnvhash`, but if my understanding is correct, I would not want to make `python3-fnvhash` a DEPENDS for `python3-fnv-hash-fast` if it is not going to ever use it because it will bloat the disk space footprint required to install it.

What are your throughts?

Reply:
That's my understanding from the README as well, but since Cython is a [build requirement](https://github.com/bdraco/fnv-hash-fast/blob/v0.5.0/pyproject.toml#L82), the CPP implementation of fnv1a (32) should always be built and fnvhash never used, and yet fnvhash is still listed as a runtime dependency. Perhaps this can be clarified/fixed upstream.

Reply:
I'll review the code and see what it's actually doing.

Reply:
Yes, I can confirm that `fnvhash` it is required. Something had installed `fnvhash` using pip automatically and I missed it.

Live testing confirms:

```
root@storage:~# ls /mnt/storage/python3.11/site-packages/fnvhash*
fnvhash-0.1.0.dist-info/  fnvhash/
root@storage:~# mv /mnt/storage/python3.11/site-packages/fnvhash /tmp/
root@storage:~# python
Python 3.11.6 (main, Oct 30 2023, 23:44:03) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fnv_hash_fast
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.11/site-packages/fnv_hash_fast/__init__.py", line 3, in <module>
  File "/usr/lib/python3.11/site-packages/fnv_hash_fast/fnv_impl.py", line 4, in <module>
ModuleNotFoundError: No module named 'fnvhash'
```
... as well as the actual [code](https://github.com/bdraco/fnv-hash-fast/blob/main/src/fnv_hash_fast/fnv_impl.py):

```
from fnvhash import FNV1_32_INIT, FNV_32_PRIME, fnva

_FNV_SIZE = 2**32

if TYPE_CHECKING:
    fnv1a_32: Callable[[bytes], int]

fnv1a_32 = partial(
    fnva, hval_init=FNV1_32_INIT, fnv_prime=FNV_32_PRIME, fnv_size=_FNV_SIZE
)

try:
    from ._fnv_impl import (  # type: ignore[no-redef] # noqa: F811 F401
        _fnv1a_32 as fnv1a_32,
    )
except ImportError:
    pass
```